enterprise/server/githubapp/githubapp.go

```diff
@@ -1472,6 +1473,69 @@ func (a *GitHubApp) populatePRMetadata(ctx context.Context, client *github.Clien
 	return prs, nil
 }
 
+func (a *GitHubApp) cachedUser(ctx context.Context, client *github.Client) (*github.User, error) {
+	key := "user"
+	pr, err := a.cached(ctx, key, &github.User{}, time.Hour*24, func() (any, any, error) {
+		return client.Users.Get(ctx, "")
+	})
+	return pr.(*github.User), err
+}
+
+func (a *GitHubApp) cachedSearch(ctx context.Context, client *github.Client, query string) (*github.IssuesSearchResult, error) {
+	key := fmt.Sprintf("search/%s", query)
+	pr, err := a.cached(ctx, key, &github.IssuesSearchResult{}, time.Second*5, func() (any, any, error) {
+		return client.Search.Issues(ctx, query, &github.SearchOptions{ListOptions: github.ListOptions{PerPage: 100}})
+	})
+	return pr.(*github.IssuesSearchResult), err
+}
+
+func (a *GitHubApp) cachedPullRequests(ctx context.Context, client *github.Client, owner, repo string, number int, updatedAt time.Time) (*github.PullRequest, error) {
+	key := fmt.Sprintf("pr/%s/%s/%d/%d", owner, repo, number, updatedAt.Unix())
+
+	pr, err := a.cached(ctx, key, &github.PullRequest{}, time.Hour*24, func() (any, any, error) {
+		return client.PullRequests.Get(ctx, owner, repo, number)
+	})
+	return pr.(*github.PullRequest), err
+}
+
+func (a *GitHubApp) cachedReviews(ctx context.Context, client *github.Client, owner, repo string, number int, updatedAt time.Time) (*[]*github.PullRequestReview, error) {
+	key := fmt.Sprintf("pr-reviews/%s/%s/%d/%d", owner, repo, number, updatedAt.Unix())
+	pr, err := a.cached(ctx, key, &[]*github.PullRequestReview{}, time.Hour*24, func() (any, any, error) {
+		rev, res, err := client.PullRequests.ListReviews(ctx, owner, repo, number, &github.ListOptions{PerPage: 100})
+		return &rev, res, err
+	})
+	return pr.(*[]*github.PullRequestReview), err
+}
+
+func (a *GitHubApp) cached(ctx context.Context, key string, v any, exp time.Duration, fn func() (any, any, error)) (any, error) {
+	tu, err := a.env.GetUserDB().GetUser(ctx)
```

Comment:
(I honestly don't know if this is the right thing to use anymore -- seems like it may be doing more work than is necessary?)